packages/ra-core/src/auth/useGetIdentity.ts

```diff
@@ -38,42 +41,61 @@ const defaultIdentity = {
  *     }
  * }
  */
-const useGetIdentity = () => {
-    const [state, setState] = useSafeSetState<State>({
-        isLoading: true,
-    });
+export const useGetIdentity = (
+    queryParams: UseQueryOptions<UserIdentity, Error> = defaultQueryParams
+): UseGetIdentityResult => {
     const authProvider = useAuthProvider();
-    useEffect(() => {
-        if (authProvider && typeof authProvider.getIdentity === 'function') {
```

Comment:
Removing typeof authProvider.getIdentity === 'function' check causes a tiny BC break.

We found there are console errors after upgrading pass 4.6 because we have never implemented the getIdentity method since it was introduced in https://github.com/marmelab/react-admin/pull/5180 

In the previous pull request it says 
> The change is backward compatible (if the authProvider.getIdentity() method isn't implemented, the admin shows an anonymous user avatar with no name, as before).

In v4 upgrade guide it did not mention this method is now required to be implemented. I believe we should now add a note about that breaking change.

Reply:
It will be very easy for us to catch the issue and implement the method.

Just for other users they should be aware when they upgrade.

Reply:
Thanks for pointing that out!
The `getIdentity` function should remain optional, even in v4, so I believe we should reintroduce this check to avoid errors. I'll open a new PR.

Reply:
Here it is: https://github.com/marmelab/react-admin/pull/8509